### PR TITLE
feat: implement `size_hint` on `ResultIterator`

### PR DESCRIPTION
**Pull request by jhgg**
we know the row count, so we can implement `size_hint` on `ResultIterator`, which will let various `collect` impl's be slightly more efficient.

### CODE REVIEW

File: src/cassandra/result.rs [hunk 1]
```diff
@@ -193,6 +202,10 @@ impl<'a> Iterator for ResultIterator<'a> {
             }
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (self.1, Some(self.1))
```
Comment (kw217) [inline src/cassandra/result.rs:207]: The lower bound isn't correct - after we've called `next()` the lower bound will be smaller. We should either update `self.1` in `next`, or put `0` here.

**Review by kw217 — CHANGES_REQUESTED** [review]
I like the idea, but there's a correctness issue. Happy to merge once that's fixed.

File: src/cassandra/result.rs (hunk 1 above)
Comment (jhgg) [reply src/cassandra/result.rs:207]: good catch. i'll set to 0

Comment (kw217) [reply src/cassandra/result.rs:207]: sounds good.

**Review by kw217 — APPROVED** [review]
LGTM - thanks!

**Comment by kw217** [comment]
Hi @jhgg - I've just spotted that somehow the commit checker didn't run for one of these commit, and so I didn't notice that you didn't sign your commit per CONTRIBUTING to acknowledge the Developer Certificate of Origin. Please could you state below "I hereby certify the Developer Certificate of Origin in [CONTRIBUTING](https://github.com/Metaswitch/cassandra-rs/blob/master/CONTRIBUTING.md) with respect to my commit 741e6991393f4dcf68f7ab896b21948db3b19d5d"? Very sorry about this, but we have to ensure the driver is legally clean so it can continue to be used and distributed freely. Thanks.

**Comment by jhgg** [comment]
I hereby certify the Developer Certificate of Origin in CONTRIBUTING with respect to my commit 741e699

**Comment by kw217** [comment]
Great, thanks @jhgg!

**Pull request merged**